plugins/MauticCrmBundle/Api/DynamicsApi.php

```diff
@@ -222,7 +222,7 @@ public function parseRawHttpResponse(ResponseInterface $response)
     {
         $a_data      = [];
         $input       = $response->getBody();
-        $contentType = $response->getHeaders()['Content-Type'];
+        $contentType = $response->getHeaders()['Content-Type'][0];
```

Comment:
I'm afraid it is not certain that it returns an array. Can you make it work for string as well as array? And it will need to be covered with tests to get this merged.

Reply:
`    /**
     * Retrieves all message header values.
     *
     * The keys represent the header name as it will be sent over the wire, and
     * each value is an array of strings associated with the header.
     *
     *     // Represent the headers as a string
     *     foreach ($message->getHeaders() as $name => $values) {
     *         echo $name . ": " . implode(", ", $values);
     *     }
     *
     *     // Emit headers iteratively:
     *     foreach ($message->getHeaders() as $name => $values) {
     *         foreach ($values as $value) {
     *             header(sprintf('%s: %s', $name, $value), false);
     *         }
     *     }
     *
     * While header names are not case-sensitive, getHeaders() will preserve the
     * exact case in which headers were originally specified.
     *
     * @return string[][] Returns an associative array of the message's headers. Each
     *     key MUST be a header name, and each value MUST be an array of strings
     *     for that header.
     */
    public function getHeaders();`

it returns an array of headers with in it an array of strings, cannot find anywhere that this can return a single string (only in the getHeaderLine call), might you be confused with an older/wrong implementation of this interface?

Reply:
Seems there was a http guzzle refactor, this made it go PSR7, other parts of mautic bugged at the time (For example: https://github.com/mautic/mautic/pull/10308 ). The Dynamics plugin had the same issue, but this was not catched at the time.

Do you expect me to write PHPunit testing for this whole plugin? As it is non-existent.

Reply:
You are right. The `Psr\Http\Message\ResponseInterface::getHeaders()` documents well that the header value is an array. And thanks for getting to the bottom of what changed this behavior.

The test should cover the `parseRawHttpResponse` method by sending an example response to it. The more cases you can cover the better but it must cover the `'Content-Type'` header and work with it.